### ISSUE TITLE
Build fails, unknown cause, little output...

### ISSUE THREAD

**Issue by hrdwdmrbl** [steemit/steem#617]
```
[  1%] Built target equihash
[  7%] Built target udt
[ 11%] Built target project_secp256k1
[ 11%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/thread/thread.cpp.o
c++: internal compiler error: Killed (program cc1plus)
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-4.9/README.Bugs> for instructions.
make[2]: *** [libraries/fc/CMakeFiles/fc.dir/src/thread/thread.cpp.o] Error 4
make[1]: *** [libraries/fc/CMakeFiles/fc.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by mvandeberg** [comment]
@hrdwdmrbl You have made several issues about failing builds. I am consolidating your issues here. Can you please post your build environment? (gcc version, cmake version, boost version, etc.)

#615
> Followed instructions [here](https://steem.io/documentation/how-to-build/) but they don't work on Ubuntu 14.04
> 
> ```
> [ 12%] Building CXX object libraries/fc/CMakeFiles/fc.dir/src/interprocess/process.cpp.o
> In file included from /usr/lib/x86_64-linux-gnu/include/boost/iostreams/detail/is_dereferenceable.hpp:12:0,
>                  from /usr/lib/x86_64-linux-gnu/include/boost/iostreams/detail/resolve.hpp:26,
>                  from /usr/lib/x86_64-linux-gnu/include/boost/iostreams/detail/push.hpp:24,
>                  from /usr/lib/x86_64-linux-gnu/include/boost/iostreams/detail/streambuf/indirect_streambuf.hpp:31,
>                  from /usr/lib/x86_64-linux-gnu/include/boost/iostreams/stream_buffer.hpp:22,
>                  from /usr/lib/x86_64-linux-gnu/include/boost/iostreams/stream.hpp:21,
>                  from /root/blockchains/steemit/steem/libraries/fc/src/interprocess/process.cpp:8:
> /usr/lib/x86_64-linux-gnu/include/boost/type_traits/detail/bool_trait_def.hpp:18:79: note: #pragma message: NOTE: Use of this header (bool_trait_def.hpp) is deprecated
>  # pragma message("NOTE: Use of this header (bool_trait_def.hpp) is deprecated")
>                                                                                ^
> In file included from /usr/lib/x86_64-linux-gnu/include/boost/type_traits/detail/bool_trait_def.hpp:21:0,
>                  from /usr/lib/x86_64-linux-gnu/include/boost/iostreams/detail/is_dereferenceable.hpp:12,
>                  from /usr/lib/x86_64-linux-gnu/include/boost/iostreams/detail/resolve.hpp:26,
>                  from /usr/lib/x86_64-linux-gnu/include/boost/iostreams/detail/push.hpp:24,
>                  from /usr/lib/x86_64-linux-gnu/include/boost/iostreams/detail/streambuf/indirect_streambuf.hpp:31,
>                  from /usr/lib/x86_64-linux-gnu/include/boost/iostreams/stream_buffer.hpp:22,
>                  from /usr/lib/x86_64-linux-gnu/include/boost/iostreams/stream.hpp:21,
>                  from /root/blockchains/steemit/steem/libraries/fc/src/interprocess/process.cpp:8:
> /usr/lib/x86_64-linux-gnu/include/boost/type_traits/detail/template_arity_spec.hpp:13:84: note: #pragma message: NOTE: Use of this header (template_arity_spec.hpp) is deprecated
>  # pragma message("NOTE: Use of this header (template_arity_spec.hpp) is deprecated")
>                                                                                     ^
> In file included from /usr/lib/x86_64-linux-gnu/include/boost/iostreams/detail/is_dereferenceable.hpp:13:0,
>                  from /usr/lib/x86_64-linux-gnu/include/boost/iostreams/detail/resolve.hpp:26,
>                  from /usr/lib/x86_64-linux-gnu/include/boost/iostreams/detail/push.hpp:24,
>                  from /usr/lib/x86_64-linux-gnu/include/boost/iostreams/detail/streambuf/indirect_streambuf.hpp:31,
>                  from /usr/lib/x86_64-linux-gnu/include/boost/iostreams/stream_buffer.hpp:22,
>                  from /usr/lib/x86_64-linux-gnu/include/boost/iostreams/stream.hpp:21,
>                  from /root/blockchains/steemit/steem/libraries/fc/src/interprocess/process.cpp:8:
> /usr/lib/x86_64-linux-gnu/include/boost/type_traits/detail/template_arity_spec.hpp:13:84: note: #pragma message: NOTE: Use of this header (template_arity_spec.hpp) is deprecated
>  # pragma message("NOTE: Use of this header (template_arity_spec.hpp) is deprecated")
>                                                                                     ^
> c++: internal compiler error: Killed (program cc1plus)
> Please submit a full bug report,
> with preprocessed source if appropriate.
> See <file:///usr/share/doc/gcc-4.8/README.Bugs> for instructions.
> make[2]: *** [libraries/fc/CMakeFiles/fc.dir/src/interprocess/process.cpp.o] Error 4
> make[1]: *** [libraries/fc/CMakeFiles/fc.dir/all] Error 2
> make: *** [all] Error 2
> ```

**Comment by abitmore** [comment]
@mvandeberg: as mentioned here (https://github.com/steemit/steem/issues/619#issuecomment-263175528) this issue should be caused by insufficient RAM. I think it's safe to close it.

By the way, perhaps it's good to add a "system requirements" section or so in readme.md and/or build instructions.

**Comment by hrdwdmrbl** [comment]
@mvandeberg 
> You have made several issues about failing builds. I am consolidating your issues here.

Yes, sorry about the noise. They were all separate though because almost each time I encountered one of them, I would reset my VM to base image and try again. :)
At all times I tried to follow the instructions on the [here](https://github.com/steemit/steem/blob/develop/doc/building.md) or [here](https://steem.io/documentation/how-to-build/)

I tried following with 14.04 and 16.04 versions of Ubuntu.

@abitmore 
> his issue should be caused by insufficient RAM. I think it's safe to close it.

Yes, that's almost certainly one of my problems, though I'm not sure it's the entire problem because I often was not compiling with the `-g`. The machine I was trying on was only a 512mb machine though :)

> I think it's safe to close it.

You're probably right, especially because I wiped each of the machines I had problems on and can no longer held with a debugging process. I half just created them in case there was a solution or simple fix, and half to document the issues in case other users encounter the same issue.

> By the way, perhaps it's good to add a "system requirements" section or so in readme.md and/or build instructions.

This sounds like a great idea. I would also consolidate [this](https://github.com/steemit/steem/blob/develop/doc/building.md) or [thishttps://steem.io/documentation/how-to-build/) and then put the Sys Reqs in the consolidated document.
